### PR TITLE
Statistics tracker improvements

### PR DESCRIPTION
**Pull request by sschaef**
See commit messages for more details.

### CODE REVIEW

**Comment by dragos** [comment]
Incrementing an in-memory counter shouldn't take too much time. In fact, it might be way slower now, since the overhead of a Job is quite significant. My remark previously was about loading the Statistics file, which involves disk access, but if that is done only once, on startup, I think it's all fine.

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/statistics/Statistics.scala [hunk 1]
```diff
@@ -16,9 +21,12 @@ import Features._
 /**
  * Contains all definitions that belong to the statistics tracker.
  */
-class Statistics {
+class Statistics extends AnyRef with HasLogger {
```
Comment (wpopielarski) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/statistics/Statistics.scala]: just curious about extending AnyRef, any hidden scala feature?

**Comment by wpopielarski** [comment]
actually LGTM

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/statistics/Statistics.scala (hunk 1 above)
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/statistics/Statistics.scala]: No hidden feature. I just don't like to use `extend` with a trait. Unnecessary thoughts...

**Comment by sschaef** [comment]
On 12/01/2015 10:36 AM, Iulian Dragos wrote:
>
> Incrementing an in-memory counter shouldn't take too much time. In 
> fact, it might be way slower now, since the overhead of a Job is quite 
> significant. My remark previously was about loading the Statistics 
> file, which involves disk access, but if that is done only once, on 
> startup, I think it's all fine.
>
Indeed, now after you said it I'm going to remove that commit. I thought 
that because we not only increment but also write back the chances to 
disk, it could take some time. But some benchmarks showed that is still 
doesn't even need 1ms on a SSD. Don't know how huge the slowdown is for 
a HDD but probably negligible.

**Pull request merged**